### PR TITLE
multiple genomes pointing to the same assembly in Enigma

### PR DESCRIPTION
**Pull request by Xiangs18**
This PR address [this comment](https://kbase.slack.com/archives/C03FK9RKSBX/p1685660960874599)

### CODE REVIEW

File: src/loaders/workspace_downloader/workspace_downloader.py [hunk 1]
```diff
@@ -185,13 +190,23 @@ def _assembly_genome_lookup(genome_objs):
...
-            raise ValueError(
-                f"Multiple genomes match to the assembly upa {assembly_upa}"
-            )
+        genome_old = hashmap.get(assembly_upa)
+        id_and_date = [genome_upa, save_date]
+        if not genome_old:
+            hashmap[assembly_upa] = id_and_date
+            continue
+
+        if genome_old[1] > save_date:
```
Comment (MrCreosote) [inline src/loaders/workspace_downloader/workspace_downloader.py:199]:
```suggestion
        elif genome_old[1] > save_date:
```

File: src/loaders/workspace_downloader/workspace_downloader.py [hunk 2]
```diff
@@ -414,7 +429,10 @@ def main():
             conf,
             loader_common_names.OBJECTS_NAME_ASSEMBLY,
         )
-        assembly_genome_map = _assembly_genome_lookup(genome_objs)
+        assembly_genome_map, duplicate_map = _assembly_genome_lookup(genome_objs)
+        if duplicate_map:
+            with open(os.path.join(output_dir, GENOME_DUPLICATE_FILE), "w") as outfile:
```
Comment (MrCreosote) [inline src/loaders/workspace_downloader/workspace_downloader.py:434]: At this point could you
* sort each of the lists in the missing dict by the save date
* print out a message saying something along the lines of that duplicates were found, the latest was kept, and the duplicates are in a file at `<path>`

File: src/loaders/workspace_downloader/workspace_downloader.py (hunk 1 above)
Comment (Xiangs18) [reply src/loaders/workspace_downloader/workspace_downloader.py]: Updated!

File: src/loaders/workspace_downloader/workspace_downloader.py (hunk 2 above)
Comment (Xiangs18) [reply src/loaders/workspace_downloader/workspace_downloader.py]: Updated. Currently is in ascending order.

File: src/loaders/workspace_downloader/workspace_downloader.py (hunk 1 above)
Comment (MrCreosote) [reply src/loaders/workspace_downloader/workspace_downloader.py]: :+1:

File: src/loaders/workspace_downloader/workspace_downloader.py [hunk 3]
```diff
@@ -414,7 +427,18 @@ def main():
...
+        if duplicate_map:
+            for assembly_upa, id_and_date in duplicate_map.items():
+                duplicate_map[assembly_upa] = sorted(id_and_date, key=lambda x: x[1])
+
+            duplicate_path = os.path.join(output_dir, GENOME_DUPLICATE_FILE)
+            with open(duplicate_path, "w") as outfile:
+                json.dump(duplicate_map, outfile)
+
+            print(
+                f"Duplicates were found, only the latest was kept, and the duplicates are in a file at {duplicate_path}"
```
Comment (MrCreosote) [inline src/loaders/workspace_downloader/workspace_downloader.py:440]: Duplicates of what?

File: src/loaders/workspace_downloader/workspace_downloader.py (hunk 2 above)
Comment (MrCreosote) [reply src/loaders/workspace_downloader/workspace_downloader.py]: :+1:

File: src/loaders/workspace_downloader/workspace_downloader.py (hunk 3 above)
Comment (MrCreosote) [reply src/loaders/workspace_downloader/workspace_downloader.py]: :+1:

**Review by Tianhao-Gu — APPROVED** [review]
👍

**Pull request merged**